LoopKitUI/Views/ExpandableDatePicker.swift

```diff
@@ -0,0 +1,63 @@
+//
+//  ExpandableDatePicker.swift
+//  LoopKitUI
+//
+//  Created by Anna Quinlan on 8/12/20.
+//  Copyright © 2020 LoopKit Authors. All rights reserved.
+//
+
+import SwiftUI
+
+public struct ExpandableDatePicker: View {
+    @State var dateShouldExpand = false
+    @Binding var date: Date
+    let placeholderText: String
+    let pickerRange: ClosedRange<Date>
+    @State public var userDidTap: Bool = false
+    
+    public init (
+        with date: Binding<Date>,
+        pickerRange: ClosedRange<Date>? = nil,
+        placeholderText: String = ""
+    ) {
+        _date = date
+        self.placeholderText = placeholderText
+        
+        let today = Date()
+        self.pickerRange = pickerRange ?? today.addingTimeInterval(-.hours(24))...today
+    }
+    
+    public var body: some View {
+        VStack(spacing: 0) {
+            HStack {
+                dateFieldText
+                Spacer()
+            }
+            .padding()
+            .frame(minWidth: 0, maxWidth: .infinity).onTapGesture {
+                self.userDidTap = true
```

Comment:
If user hasn't tapped the cell, then we should show the placeholder text; otherwise, display the date in the picker